eslint.config.mjs

```diff
@@ -1,9 +1,24 @@
 import js from '@eslint/js';
 import mocha from 'eslint-plugin-mocha';
 export default [
-  mocha.configs.flat.recommended,
```

Comment:
what happened here?

is this because we weren't running lint before? 🤔 

if we do need this, better to specify your own object rather than merging with mocha's:

```json
{
  languageOptions: {
    globals: {
      ...
    }
  }
}
```

it'll apply to all files other objects already matched

Reply:
i don't want to apply new globals outside of the tests, as describe/it/etc. are specific to tests, which is what the mocha configs apply to.

i'm not sure what the exact desired process was, but they weren't picked up in my lints before at all until i modified this config.

Reply:
to keep things simple, i would just specify a separate object rather than having to spread each part of mocha's config:

```ts
[
  mocha.configs.flat.recommended,
  {
    files: ["test/**/*.js"],
    languageOptions: {
      globals: {
        /* various globals */
      }
    }
  }
]
```

Reply:
ok, i can go simplify this 👍

Reply:
while i'm doing this, there's an issue on the `param` method where name and value aren't used in http.js line 384, according to eslint... and they're definitely not used.

would you prefer i remove them and fix your jsdoc? i'm happy to do it in another PR, if preferred.

Reply:
given that the jsdoc won't be inferred in tooling (i.e.  we're documenting a method no user will ever see directly, the jsdoc for chai itself won't inherit this afaik), i think we're good to remove the params